rclpy/src/rclpy/_rclpy.c

```diff
@@ -3660,6 +3668,18 @@ rclpy_convert_from_py_qos_policy(PyObject * Py_UNUSED(self), PyObject * args)
   qos_profile->depth = pyqos_depth;
   qos_profile->reliability = pyqos_reliability;
   qos_profile->durability = pyqos_durability;
+
+  qos_profile->lifespan.sec = RCUTILS_NS_TO_S(pyqos_lifespan);
+  qos_profile->lifespan.nsec = pyqos_lifespan - RCUTILS_NS_TO_S(qos_profile->lifespan.sec);
```

Comment:
```suggestion
  qos_profile->lifespan.nsec = pyqos_lifespan - RCUTILS_S_TO_NS(qos_profile->lifespan.sec);
```

You need to go from Sec to NSec. Looks like this is already correct on the other lines below.

Reply:
Thanks - this is why we want to force each other to write tests for everything